### PR TITLE
Add test for rare RB init edge case

### PR DESCRIPTION
**Pull request by nyanzebra**
It is possible for a crash to occur where there are no deleted block but reads have occurred. In this case we would want to have read at write position on init as this should only happen if the buffer is full.

### CODE REVIEW

File: mqtt/mqtt-bridge/src/persist/waking_state/ring_buffer/mod.rs [hunk 1]
```diff
@@ -865,6 +870,85 @@ mod tests {
         }
     }
 
+    #[test]
+    fn it_inits_ok_with_previous_data_and_write_pointer_is_reaches_read_pointer_after_read() {
+        let result = tempfile::NamedTempFile::new();
+        assert_matches!(result, Ok(_));
```
Comment (vadim-kovalyov) [inline mqtt/mqtt-bridge/src/persist/waking_state/ring_buffer/mod.rs:876]: I'm not sure why do you need this? why not just use `.expect("...")` or `.unwrap()` directly? I believe asserts are better for actual test results you want to verify, not for intermediate/setup code.

Comment (nyanzebra) [reply mqtt/mqtt-bridge/src/persist/waking_state/ring_buffer/mod.rs:876]: Yeah will change :)

File: mqtt/mqtt-bridge/src/persist/waking_state/ring_buffer/mod.rs [hunk 2]
```diff
@@ -865,6 +870,80 @@ mod tests {
...
+                RingBuffer::new(&file.path().to_path_buf(), max_file_size, FLUSH_OPTIONS).unwrap();
+
+            // write some
+            for _ in 0..10 {
+                let result = rb.insert(&publication);
+                assert_matches!(result, Ok(_));
+            }
+
+            let result = rb.batch(10);
+            assert_matches!(result, Ok(_));
```
Comment (vadim-kovalyov) [inline mqtt/mqtt-bridge/src/persist/waking_state/ring_buffer/mod.rs:907]: I think this can also be replaced with `.expect(..)` or `.unwrap()`.

File: mqtt/mqtt-bridge/src/persist/waking_state/ring_buffer/mod.rs [hunk 3]
```diff
@@ -865,6 +870,80 @@ mod tests {
...
+            let mut batch = result.unwrap();
+            assert!(!batch.is_empty());
+
+            for (key, _) in batch.drain(..) {
+                let result = rb.remove(key);
```
Comment (vadim-kovalyov) [inline mqtt/mqtt-bridge/src/persist/waking_state/ring_buffer/mod.rs:912]: instead of `assert_matches!`
```suggestion
                let result = rb.remove(key).expect(format!("unable to remove pub with key {}", key));
```

File: mqtt/mqtt-bridge/src/persist/waking_state/ring_buffer/mod.rs (hunk 2 above)
Comment (nyanzebra) [reply mqtt/mqtt-bridge/src/persist/waking_state/ring_buffer/mod.rs]: Done :)

**Review by vadim-kovalyov — APPROVED** [review]
